parser/ParseIndexProperties.cpp

```diff
@@ -0,0 +1,28 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "parser/ParseIndexProperties.hpp"
+
+#include <string>
+
+namespace quickstep {
+  // initialize constants for various index properties
+  const std::string BloomFilterIndexProperties::kBloomFilterSizeInBytes = "size";
```

Comment:
Yes, for the performance reasons.

After I checked the usages for these strings, and `valid_property_map_` and `hasInvalidCustomProperty` defined in the base class, there could have a better OO implementation here:
 1. mark `hasInvalidCustomProperty` virtual with the default true return value, and `BloomFilterIndexProperties` has its own implementation for size in bytes, num hashes, and project element count.
 1. remove `valid_property_map_` in the base class.